services/cd-service/pkg/repository/db.go

```diff
@@ -230,6 +232,115 @@ func (h *DBHandler) DBWriteAllApplications(ctx context.Context, previousVersion
 	return nil
 }
 
+func (h *DBHandler) writeEvent(ctx context.Context, eventuuid, eventType, sourceCommitHash string, eventJson []byte) error {
+	span, ctx := tracer.StartSpanFromContext(ctx, "writeEvent")
+	defer span.Finish()
+	tx, err := h.DB.BeginTx(ctx, nil)
+	if err != nil {
+		return fmt.Errorf("could not begin transaction. Error: %w", err)
+	}
+	insertQuery := h.AdaptQuery("INSERT INTO events (uuid, timestamp, commitHash, eventType, json)  VALUES (?, ?, ?, ?, ?);")
+
+	logger.FromContext(ctx).Sugar().Warnf("Query: %s", insertQuery)
+
+	rawUUID, err := timeuuid.ParseUUID(eventuuid)
+	if err != nil {
+		return fmt.Errorf("error parsing UUID. Error: %w", err)
+	}
+	span.SetTag("query", insertQuery)
+	_, err = tx.Exec(
+		insertQuery,
+		rawUUID.String(),
+		uuid2.GetTime(&rawUUID).AsTime(),
+		sourceCommitHash,
+		eventType,
+		eventJson)
+
+	if err != nil {
+		return fmt.Errorf("Error inserting information into DB. Error: %w\n", err)
+	}
+	err = tx.Commit()
+	if err != nil {
+		return fmt.Errorf("could not commit transaction. Error: %w", err)
+	}
+	return nil
+}
+
+func (h *DBHandler) DBWriteDeploymentEvent(ctx context.Context, uuid, sourceCommitHash, email string, deployment *event.Deployment) error {
+
+	dataJson, err := json.Marshal(deployment)
+
+	if err != nil {
+		return fmt.Errorf("error converting deployment event data to json. Error: %w\n", err)
+	}
+
+	metadataJson, err := json.Marshal(event.Metadata{
+		AuthorEmail: email,
+		Uuid:        uuid,
+	})
+
+	if err != nil {
+		return fmt.Errorf("error converting deployment event metadata to json. Error: %w\n", err)
+	}
+	jsonToInsert, _ := json.Marshal(event.EventJson{
+		DataJson:     string(dataJson),
```

Comment:
I think this should be simplified to just write the whole json blob.
Then we would also need less `struct`s